articles/contributing/api-design-principles.md

```diff
@@ -0,0 +1,426 @@
+---
+title: Q# API Design Principles
+description: Q# API Design Principles
+author: cgranade
+ms.author: chgranad
+ms.date: 10/12/2018
+ms.topic: article
+uid: microsoft.quantum.contributing.api-design
+---
+
+# Q# API Design Principles
+
+## General Principles
```

Comment:
A lot of these seem a bit abstract. Maybe providing more examples would help.

Reply:
I've refactored to indicate which principles are intentionally somewhat abstract (key principles) and which ones give concrete and actionable guidance (the **DO**s and **DON'T**s, using ✅ and ⛔️ respectively in the style of the Azure SDK guidelines).